dev-libs/json_c/json_c-0.13.recipe

```diff
@@ -0,0 +1,77 @@
+SUMMARY="A JSON implementation in C"
+DESCRIPTION="JSON-C implements a reference counting object model that allows \
+you to easily construct JSON objects in C, output them as JSON formatted \
+strings and parse JSON formatted strings back into the C representation of \
+JSON objects."
+HOMEPAGE="https://github.com/json-c/json-c/wiki"
+COPYRIGHT="2009-2012, 2016-2017 Eric Haszlakiewicz
+	2004-2005, 2013 Metaparadigm Pte. Ltd.
+	2009-2012 Hewlett-Packard Development Company, L.P.
+	2008-2009 Yahoo! Inc.
+	2016 Alexandru Ardelean"
+LICENSE="MIT"
+REVISION="1"
+SOURCE_URI="https://s3.amazonaws.com/json-c_releases/releases/json-c-$portVersion.tar.gz"
+CHECKSUM_SHA256="0316780be9ad16c42d7c26b015a784fd5df4b0909fef0aba51cfb13e492ac24d"
+SOURCE_DIR="json-c-$portVersion"
+
+ARCHITECTURES="?x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+libVersion="3.0.1"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+PROVIDES="
+	json_c$secondaryArchSuffix = $portVersion
+	lib:libjson_c$secondaryArchSuffix = $libVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	json_c${secondaryArchSuffix}_devel = $portVersion
+	devel:libjson_c$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	haiku${secondaryArchSuffix}_devel
```

Comment:
Most ports don't seem to require haiku_devel in `REQUIRES_devel`.

```diff
-	haiku${secondaryArchSuffix}_devel
```

Reply:
If the port requires any of the Haiku headers or links against any of the system libraries it should require haiku_devel. This is not a question of statistics. Some core devel packages will pull it in so that it becomes available, but this doesn't mean that it's correct to leave it out.

Reply:
@mmlr 
Then I believe most `_devel` packages should include `haiku_devel`, as they use the C standard library headers in one way or another.

Given the huge number of ports not having `haiku_devel` in their `REQUIRES` for the devel package (820/860), I would like to know why wasn't this added for them?  Was it because we always add `haiku_devel` to `BUILD_REQUIRES`?

Reply:
> Was it because we always add haiku_devel to BUILD_REQUIRES?

Yep, I guess this is the reason.

Anyway, since **_all_** recipes which would need to mention `devel:libjson_c` (or `devel:libjson_c$secondaryArchSuffix` in it's `BUILD_REQUIRES` would also have to mention `haiku_devel` (or `haiku${secondaryArchSuffix}_devel`) in that same `BUILD_REQUIRES`, I think we can make one more "exception" here, i.e. allow REQUIRES_devel to **_not_** mention `haiku${secondaryArchSuffix}_devel`, as this is the case for most recipes (820/860, according to @alaviss).
It does not hurt, otherwise we would already have removed all these exceptions :)